firefox-ios/firefox-ios-tests/Tests/ClientTests/MainMenu/MainMenuMiddlewareTests.swift

```diff
@@ -35,6 +35,82 @@ final class MainMenuMiddlewareTests: XCTestCase {
         mockStore.dispatchLegacy(action)
     }
 
+    func testTapNavigateToDestination_bookmarks() {
+        let middleware = MainMenuMiddleware()
+        let mockStore = Store(
+            state: AppState(),
+            reducer: AppState.reducer,
+            middlewares: [middleware.mainMenuProvider]
+        )
```

Comment:
To maintain consistency and readability we have a `StoreTestUtility` that this middleware test can conform too. See more details on this wiki - https://github.com/mozilla-mobile/firefox-ios/wiki/Unit-Tests-Guidelines#6-testing-with-the-store

It would clean up some setup for the tests which distracts us a bit from the actual tests imo.